### PR TITLE
Propagate axios errors 236

### PR DESCRIPTION
**Pull request by byronmurg**
## Overview

When calling an *AsStream method a client needs to handle errors
raised by couch. These errors can include retrieving an attachment
that doesn't exist or querying a deleted view.

## Testing recommendations

The easiest way to test is to try retrieving an attachment that doesn't exist.

e.g.
```
alice.attachment.getAsStream('rabbit', 'doesntexist.png')
  .on('error', (e) => console.error(e))
  .pipe(/* handle stream */)
```

## GitHub issue number

Fixes issue #236.

There are likely other issues that are affected by this, but 236 is the most recent.

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [x] Documentation reflects the changes;

The material changes are covered by the tests however I am unable to get nock errors to emit with isAxiosError.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -355,7 +355,9 @@ nano.db.list().then((body) => {
 Lists all the CouchDB databases as a stream:
 
 ```js
-nano.db.list().pipe(process.stdout);
+nano.db.list()
```
Comment (glynnbird) [inline README.md:358]: I think this should be `nano.db.listAsStream()`. An error in the original, not in your PR.

File: README.md [hunk 2]
```diff
@@ -623,7 +625,9 @@ alice.list({include_docs: true}).then((body) => {
 List all the docs in the database as a stream.
 
 ```js
-alice.list().pipe(process.stdout)
+alice.list()
```
Comment (glynnbird) [inline README.md:628]: this should be `alice.listAsStream()` - an error in the original.

File: README.md [hunk 3]
```diff
@@ -953,7 +965,9 @@ const params = {
   endkey: 'b',
   limit: 1
 }
-db.partitionedView('canidae', 'view-ddoc', 'view-name', params).pipe(process.stdout)
+db.partitionedView('canidae', 'view-ddoc', 'view-name', params)
```
Comment (glynnbird) [inline README.md:968]: This should be `db.partitionedViewAsStream(...` - an. error in the original

File: README.md [hunk 4]
```diff
@@ -1100,7 +1116,9 @@ alice.view('characters', 'happy_ones', { include_docs: true }).then((body) => {
 Same as `db.view` but returns a stream:
 
 ```js
-alice.view('characters', 'happy_ones', {reduce: false}).pipe(process.stdout);
+alice.view('characters', 'happy_ones', {reduce: false})
```
Comment (glynnbird) [inline README.md:1119]: This should be `alice.viewAsStream(...` - error in the original

File: lib/nano.js [hunk 5]
```diff
@@ -189,6 +189,33 @@ module.exports = exports = function dbScope (cfg) {
     }
   }
 
+  const streamResponseHandler = function (response, req, stream) {
```
Comment (glynnbird) [inline lib/nano.js:192]: really like this 👍

**Review by glynnbird — CHANGES_REQUESTED** [review]
Great work. Thanks for taking the time to look at this. If you could fix up the `...AsStream` documentation errors too that would be great. If you don't have time, let me know and I'll fix them separately.

File: README.md (hunk 1 above)
Comment (byronmurg) [reply README.md]: I missed these too. Corrected in the latest commit.

File: lib/nano.js (hunk 5 above)
Comment (byronmurg) [reply lib/nano.js:192]: Thanks.

**Review by glynnbird — APPROVED** [review]
Thanks!

**Pull request merged**